pints/_mcmc/__init__.py

```diff
@@ -1100,3 +1100,80 @@ def mcmc_sample(log_pdf, chains, x0, sigma0=None, method=None):
     """
     return MCMCController(    # pragma: no cover
         log_pdf, chains, x0, sigma0, method=method).run()
+
+
+def sample_initial_points(log_pdf, n_points, random_sampler=None,
+                          max_tries=None, parallel=False, n_workers=None):
+    """
+    Draws parameter values from a given sampling distribution until either
+    finite values for each of ``n_points`` have been generated or the total
+    number of attempts exceeds ``max_tries``.
+
+    If ``log_pdf`` is of :class:`LogPosterior`, then the
+    ``log_pdf.log_prior().sample`` method is used for initialisation, although
+    this is overruled by ``random_sampler`` if it is supplied.
+
+    Parameters
+    ----------
+    log_pdf : pints.LogPDF
+        A :class:`LogPDF` function that evaluates points in the parameter
+        space. It is optional that ``log_pdf`` is a of type
+        :class:`LogPosterior`.
+    n_points : int
+        The number of initial values to generate.
+    random_sampler : stochastic function
+        A function that when called returns draws from a probability
+        distribution of the same dimensionality as ``log_pdf``. The only
+        argument to this function should be an integer specifying the number of
+        draws.
+    max_tries : int
+        Number of attempts to find a finite initial value across all
+        ``n_points``. By default this is 50 x n_points.
+    parallel : Boolean
+        Whether to evaluate ``log_pdf`` in parallel (defaults to False).
+    n_workers : int
+        Number of workers on which to run parallel evaluation.
+    """
+    if random_sampler is not None and not callable(random_sampler):
+        raise ValueError("random_sampler must be a callable function.")
+
+    if random_sampler is None:
+        if isinstance(log_pdf, pints.LogPosterior):
+            random_sampler = log_pdf.log_prior().sample
+        else:
+            raise ValueError("If log_pdf not of class pints.LogPosterior " +
+                             "then random_sampler must be supplied.")
+
+    if n_points < 1:
+        raise ValueError("Number of initial points must be 1 or more.")
+
+    if max_tries is None:
```

Comment:
Just set ``max_tries=50`` in the constructor above?

Reply:
But it's not 50 tries total? It's 50 for each of the points. So I can't see a way to specify that in the constructor.

Reply:
I mean in the arguments list to __init__.

This:

```
__init__(x=None):
    if x is None:
        x=50
```

is just a long-winded way of writing

```
__init__(x=50):
```